src/__internal__/label/label.component.tsx

```diff
@@ -159,6 +161,11 @@ export const Label = ({
   return (
     <StyledLabelContainer
       data-role="label-container"
+      id={
```

Comment:
```suggestion
      id={`label-container-${labelId ?? guid.current}`}
```
completely non-blocking what you have already is fine the developer in me just can't help myself here